src/deepsparse/loggers/build_logger.py

```diff
@@ -204,6 +293,36 @@ def build_system_loggers(
     return system_loggers
 
 
+def possibly_modify_target_identifiers(
+    data_logging_config: Optional[Dict[str, List[MetricFunctionConfig]]] = None,
+    pipeline_identifier: str = None,
+) -> Optional[Dict[str, List[MetricFunctionConfig]]]:
+    """
+    Modify the target identifiers in the data logging config, given
+    the specified pipeline identifier.
+
+    :param data_logging_config: The configuration of the data loggers.
+        Specified as a dictionary that maps a target name to a list of metric functions.
+    :param pipeline_identifier: An optional string, that specifies
+        the name of the pipeline the logging is being performed for.
+    :return: the modified data_logging_config
+    """
+    if not data_logging_config or not pipeline_identifier:
+        # if either of the arguments is None, return the original config
+        return data_logging_config
+
+    for target_identifier, metric_functions in data_logging_config.copy().items():
+        if not target_identifier.startswith(pipeline_identifier):
+            # if the target identifier does not already start
+            # with the pipeline identifier, call get_target_identifier
+            # to prepend it
+            target_identifier = get_target_identifier(
+                target_identifier, pipeline_identifier
+            )
+        data_logging_config[target_identifier] = metric_functions
```

Comment:
This is a buggy implementation (will log both `some_target` and `pipeline/some_target`. Will fix it asap.